app/src/main/java/com/framgia/soundcloud/service/MusicService.java

```diff
@@ -225,15 +229,21 @@ public void setLoop(@LoopMode int loop) {
         }
     }
 
+    @Override
+    public void onBufferingUpdate(MediaPlayer mp, int percent) {
+        double ratio = percent / 100.0;
+        mBufferingLevel = (int) (mp.getDuration() * ratio);
+    }
+
     /**
      * loop mode
      */
     @Retention(SOURCE)
     @IntDef({ LoopMode.NONE_LOOP, LoopMode.LOOP_LIST_TRACKS, LoopMode.LOOP_TRACK })
     public @interface LoopMode {
-    int NONE_LOOP = 0;
-    int LOOP_LIST_TRACKS = 1;
-    int LOOP_TRACK = 2;
+        int NONE_LOOP = 0;
+        int LOOP_LIST_TRACKS = 1;
+        int LOOP_TRACK = 2;
```

Comment:
ERROR: 'member def type' have incorrect indentation level 8, expected level should be 4.